examples/temporaries.kl

```diff
@@ -0,0 +1,26 @@
+#lang "n-ary-app.kl"
+
+(import "lispy-do.kl")
+(import "list-datatype.kl")
+(import "pair-datatype.kl")
+(import "defun.kl")
+
+
+(define make-temporary
```

Comment:
What is the point of make-temporary?

Reply:
It is part of #47.

 The point is to get a fresh name that can't shadow any existing name without having to make a trampoline through the macro expander, which can be nice when programmatically generating an unknown number of binders. The alternative is a CPS'd macro that generates them and then continues with them, but that becomes too hard to follow.

See `generate-temporaries` in Scheme and Racket - this is the unary case of that.

Reply:
Okay, thanks for the reference!